src/components/markdown-renderer/hooks/use-extract-first-headline.ts

```diff
@@ -4,18 +4,24 @@
  * SPDX-License-Identifier: AGPL-3.0-only
  */
 
-import React, { useCallback, useEffect } from 'react'
+import React, { useCallback, useEffect, useRef } from 'react'
 
 export const useExtractFirstHeadline = (documentElement: React.RefObject<HTMLDivElement>, content: string, onFirstHeadingChange?: (firstHeading: string | undefined) => void): void => {
-  const extractInnerText = useCallback((node: ChildNode): string => {
+  const extractInnerText = useCallback((node: ChildNode | null): string | undefined => {
+    if (!node) {
+      return undefined
+    }
+
     let innerText = ''
 
     if ((node as HTMLElement).classList?.contains("katex-mathml")) {
       return ''
     }
 
     if (node.childNodes && node.childNodes.length > 0) {
-      node.childNodes.forEach((child) => { innerText += extractInnerText(child) })
+      node.childNodes.forEach((child) => {
+        innerText += extractInnerText(child)
```

Comment:
This could possibly create strings with the word `undefined` inside, couldn't it?

Reply:
According to the types, no. `textContent` can be null, but not undefined.

Reply:
corrected